### PR TITLE
WIP: Bootstrap plone6

### PR DESCRIPTION
**Pull request by rnixx**

### CODE REVIEW

File: src/yafowil/plone/config.py [hunk 1]
```diff
@@ -38,12 +39,16 @@
...
 ###############################################################################
 
-def configure_factory():
+def configure():
+    # configure bootstrap 4
+    bootstrap.configure_factory('bootstrap4')
+
     # set theme
-    factory.theme = 'plone5'
+    factory.theme = 'bootstrap'
```
Comment (rnixx) [inline src/yafowil/plone/config.py:47]: bootstrap or bootstrap4?

**Comment by 1letter** [comment]
@rnixx @jensens 
any news on this PR? I test today the latest es6-ready versions.cfg from buildout.coredev in a project. This project use [pas.plugins.ldap](https://github.com/collective/pas.plugins.ldap), the startup fails, but this PR [fix the issue](https://github.com/bluedynamics/yafowil.plone/pull/29/commits/1300269d35f658c5cdb895baf2a2319ace07e7a6)

**Comment by jensens** [comment]
hmm, @agitator may know more here?

**Comment by agitator** [comment]
@1letter haven't tried yet

**Comment by petschki** [comment]
@rnixx @thet @jensens @MrTango latest Plone 6.0.7 release needs latest changes because of `plone.app.widgets==5.0` can we merge and release `5.0.0a1` here ?

**Comment by 1letter** [comment]
@petschki 

If i use this branch in projects on Plone 6.0-dev (6.0.7) than the [starup error](https://github.com/bluedynamics/yafowil.plone/issues/47) is gone, but the icons for the widget are missing.
```
icon_add.png:1     Failed to load resource: the server responded with a status of 404 (Not Found) 
```
```
icon_remove.png:1     Failed to load resource: the server responded with a status of 404 (Not Found)
```
```
icon_up.png:1     Failed to load resource: the server responded with a status of 404 (Not Found)
```
```
icon_down.png:1     Failed to load resource: the server responded with a status of 404 (Not Found)
```

**Comment by rnixx** [comment]
> @rnixx @thet @jensens @MrTango Plone 6.0.7 release needs latest changes because of `plone.app.widgets==5.0` can we merge and release `5.0.0a1` here ?

sure

**Comment by petschki** [comment]
> but the icons for the widget are missing.

Sorry, I do not use the widget at all and only ran the tests after my change. But there's obvisously no test for AjaxSelectWidget implementation. Do you have a code snipped with this widget which I can test here?

**Comment by 1letter** [comment]
I use this with pas.plugin.ldap, no custom code from me. In the LDAP config form, the Icons are gone.

**Comment by petschki** [comment]
Ah thx ... I'll have a look at that.

**Comment by rnixx** [comment]
> I use this with pas.plugin.ldap, no custom code from me. In the LDAP config form, the Icons are gone.

This are the icons of the array widget https://github.com/conestack/yafowil.widget.array

**Comment by rnixx** [comment]
which version of the array widget are you using?

**Comment by 1letter** [comment]
```
mkdir -p /tmp/debug && cd /tmp/debug
touch requirements.txt
touch constraints.txt
touch mx.ini
touch instance.yaml
```
requirements.txt
``` txt
-c constraints.txt

pas.plugins.ldap
cookiecutter
```

constraints.txt
``` txt
-c https://dist.plone.org/release/6.0-dev/constraints.txt
```

mx.ini
``` ini
[settings]
requirements-in = requirements.txt
requirements-out = requirements-mxdev.txt
contraints-out = constraints-mxdev.txt

version-overrides =
    yafowil.bootstrap==2.0.0a1
    yafowil.widget.array==2.0a1
    yafowil.widget.dict==2.0a1

[yafowil.plone]
url = https://github.com/bluedynamics/yafowil.plone.git
branch=bootstrap-plone6
```

instance.yaml
``` yaml
default_context:
initial_user_name: "admin"
initial_user_password: "admin"
debug_mode: true
verbose_security: true
environment: {
  "zope_i18n_compile_mo_files": true,
  "TZ": "Europe/Berlin"
}

db_storage: "direct"
db_blobs_location: "{{ cookiecutter.location_clienthome }}/blobstorage"

```
Install Plone
```
python3 -m venv ./venv
```
```
source venv/bin/activate
```
```
pip install mxdev
```
```
pip install -r requirements-mxdev.txt
```
```
cookiecutter -f --no-input --config-file instance.yaml https://github.com/plone/cookiecutter-zope-instance
```
Start Plone
```
runwsgi -v instance/etc/zope.ini
```
Bildschirmfoto vom 2023-09-18 07-54-30

**Comment by rnixx** [comment]
The 2.x series of yafowil widgets implements webresource delivery, which is actually what we want in plone 6. But i fear yafowil.plone / yafowil.widget.* needs some love integrating the whole new resource thing properly for plone.

**Comment by petschki** [comment]
I've tried that too and before this commit 6c06d4c229a5fd971a63e4038bcba4c85c814f4d the dict widget is showing up ... that's quite strange, but I didn't have time to look deeper.

**Pull request merged**